lib/screens/boards/posts/detail/post_detail.dart

```diff
@@ -123,18 +127,21 @@ class _PostDetailState extends State<PostDetail> {
                 Padding(
                   padding: EdgeInsets.symmetric(vertical: 20),
                   child: FutureBuilder(
-                    future: context.read<Posts>().fetchComments(widget.post.id),
+                    /// FutureBuilder의 future에 명시하는 비동기 함수가 반복해서 실행되는
+                    /// 문제를 해결하고자 initState에서 정의시킨다.
+                    future: comments,
                     builder: (_, AsyncSnapshot snapshot) {
                       if (snapshot.hasData) {
                         if (snapshot.data.isNotEmpty) {
-                          // 댓글에서 중복을 허용하지 않도록 멘션 가능한 프로필 추출
                           Future.delayed(
-                            Duration.zero,
-                            () => setState(() => snapshot.data.forEach((e) {
-                              if (!mentionListId.contains(e.profile.id))
-                                mentionList.add(e.profile.toJson());
-                              mentionListId.add(e.profile.id);
-                            }))
+                            Duration.zero, () {
+                              if (this.mounted)
```

Comment:
snapshot 있을 때 duration 및 mounted 를 보는 이유가 뭐죠?

Reply:
@gajagajago 일단 **mounted**는 setState 함수 호출 시 Future.delayed 등의 비동기로 시간이 지연되는 경우 Widget이 마운트 해제되는 것을 막기 위한 안전장치입니다. 개발할 때 Simulator에서 mount 관련 에러가 매번은 아니지만 종종 뜨더라구요.
그렇다면 Duration이 어차피 zero인데 왜 **Future**로 비동기화시켰냐는 의문이 드는데요, setState를 사용하기 위해서는 Futurebuilder 내에서 비동기가 필요하다는 구글링의 산물이긴 합니다. 각종 옵션을 두고 trial-and-error 방식으로 해보기도 했는데, 위 조합이 가장 안전하다 싶어서 처리한 구문입니다. 에러메시지 전문은 아래 첨부했습니다.

```
setState() or markNeedsBuild() called during build.
This PostDetail widget cannot be marked as needing to build because the framework is already in the
process of building widgets. A widget can be marked as needing to be built during the build phase
only if one of its ancestors is currently building. This exception is allowed because the framework
builds parent widgets before children, which means a dirty descendant will always be built.
Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was:
  PostDetail
The widget which was currently being built when the offending call was made was:
  FutureBuilder<dynamic>
```